File: benchmarks/src/main/resources/js/fibonacci.array.js
function fibonacci(n) {
  if (n <= 1) {
    return n;
  }

  let fibonacci = [];
  fibonacci.push(0)
  fibonacci.push(1)
  for (let index = 2; index <= n; ++index) {
    fibonacci.push(fibonacci[index - 1] + fibonacci[index - 2])
  }

  return fibonacci[fibonacci.length - 1]
}

File: benchmarks/src/main/resources/ruby/fibonacci_array.rb
def fibonacci(n)
  if (n <= 1)
    return n
  end
  fibonacci = Array.new
  fibonacci << 0 << 1
  for index in 2..n
    fibonacci << fibonacci[index - 1] + fibonacci[index - 2]
  end
  return fibonacci[n]
end

if __FILE__ == $0
  n = ARGV[0].to_i
  puts(fibonacci(n))
end